### ISSUE TITLE
Multimodality Data Training

### ISSUE THREAD

**Issue by JunweiLiu0208** [clementchadebec/benchmark_VAE#64]
**Is your feature request related to a problem? Please describe.**

Hi, Thanks for the integrated framework for VAE learning, it works well with a single-modal dataset,
Now, I want to perform multi-modality training with benchmark_VAE but I cannot find some introduction
about custom the reconstruction loss, for instance, calculating the loss per modal and combining them into the final loss,
Can you provide some ideas about how to build a custom reconstruction loss function? 

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by clementchadebec** [comment]
Hi @JunweiLiu0208,
Thanks for opening this issue. As of today this library does not support multi-modal training in its current shape. However, this is something I am thinking of in the long run. I think this easiest way for you to do this is to fork this repo and apply some changes (such as amending the reconstruction losses) to the current code structure so you can keep the implementation of the models while making it able to handle various modalities.

The main changes you may need to make are the following:
1. Create your custom [`pythae.data.Dataset`](https://github.com/clementchadebec/benchmark_VAE/blob/main/src/pythae/data/datasets.py) so it can outputs all the modalities.
2. Amend a bit the models implementation:
2.a. the `loss_function`: with you custom reconstruction loss by modality for instance.
2.b. the `forward` method: so that it can handle multi model dataset. For instance, it will need to be able to get each modality from the inputs coming from your custom `Dataset`.
https://github.com/clementchadebec/benchmark_VAE/blob/a72b72ef5bf8cd6f1ff9f3de543e12565ee7e825/src/pythae/models/vae/vae_model.py#L78

3. Please note that you may not have to amend the `pythae.trainers` since they are not model and data dependent.

I will keep in mind this request and will start thinking of a way to allow users to define their own losses since this is something that may interest other users as well.

I hope this helps,

Best,

Clément

**Comment by JunweiLiu0208** [comment]
Thanks a lot, I will try to make it works.

**Comment by clementchadebec** [comment]
Cool! Do not hesitate if you have any questions regarding the code :smiley:

**Comment by clementchadebec** [comment]
Closed due to inactivity
